src/main.cpp

```diff
@@ -110,6 +131,42 @@ int main(int argc, char** argv)
             }
             defines.push_back(define);
         }
+        else if (a.rfind("-W", 0) == 0)
+        {
+            std::string option;
+            bool enabled;
+            if (a.rfind("-Wno-", 0) == 0)
+            {
+                enabled = false;
+                option = a.substr(5);
```

Comment:
Would this cause a SEGV if the string length was exactly 5?

Reply:
According to docs:
> If this is equal to the string length, the function returns an empty string.
The return will be an empty string if the length is 5.